src/condor_schedd.V6/qmgmt_send_stubs.cpp

```diff
@@ -125,14 +125,41 @@ NewCluster()
 		neg_on_error( qmgmt_sock->code(rval) );
 		if( rval < 0 ) {
 			neg_on_error( qmgmt_sock->code(terrno) );
-			neg_on_error( qmgmt_sock->end_of_message() );
 			errno = terrno;
+
+			// Older versions of the SCHEDD do not return an error reply ad
+			// To handle that, look at what's on the wire, rather than what we should expect.
+			ClassAd reply;
+			bool gotClassad = false;
+			if ( ! qmgmt_sock->peek_end_of_message()) {
+				gotClassad = getClassAd(qmgmt_sock, reply);
+			}
+			if ( ! qmgmt_sock->end_of_message() && ! terrno) { // prefer existing errno
+				errno = ETIMEDOUT;
```

Comment:
Setting of errno should happen right before we return. Intervening function calls may overwrite it.

Reply:
fixed